### PR TITLE
Add checks in ERC20WrapperBundler

### PR DESCRIPTION
**Pull request by Jean-Grimal**
Fixes https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/3c786edb-0c0d-4398-a1d3-a1067a935156

### CODE REVIEW

File: src/ERC20WrapperBundler.sol [hunk 1]
```diff
@@ -34,7 +34,8 @@ abstract contract ERC20WrapperBundler is BaseBundler {
         require(amount != 0, ErrorsLib.ZERO_AMOUNT);
 
         _approveMaxTo(address(underlying), wrapper);
-        ERC20Wrapper(wrapper).depositFor(initiator(), amount);
+        bool success = ERC20Wrapper(wrapper).depositFor(initiator(), amount);
+        require(success, ErrorsLib.UNSECCESSFULL_DEPOSIT);
```
Comment (MerlinEgalite) [inline src/ERC20WrapperBundler.sol:38]:
```suggestion
        require(success, ErrorsLib.DEPOSIT_FAILED);
```

It's more concise, but better to wait the feedback of others maybe

Comment (adhusson) [reply src/ERC20WrapperBundler.sol:38]: Btw if you keep the original formulation, it's `UNSUCCESSFULL_*` (s/ECC/UCC)

Comment (Rubilmax) [reply src/ERC20WrapperBundler.sol:38]: `UNSUCCESSFUL_*` 😁

**Review by Rubilmax — COMMENTED** [review]
Code looks good! Is it a real issue though? We could argue the implementation of `ERC20WrapperBundler` is not compatible with non-reverting `ERC20Wrapper` implementations and it could be fine. do we know of wrappers that don't revert and instead return `false`?

The cost of such a late change gets worse and worse as time goes... We are a couple of weeks away from deployment

**Comment by MerlinEgalite** [comment]
> Code looks good! Is it a real issue though? We could argue the implementation of ERC20WrapperBundler is not compatible with non-reverting ERC20Wrapper implementations and it could be fine. do we know of wrappers that don't revert and instead return false?

Indeed we could say that

> We are a couple of weeks away from deployment

*days btw

File: src/ERC20WrapperBundler.sol (hunk 1 above)
Comment (MathisGD) [inline src/ERC20WrapperBundler.sol:38]:
```suggestion
        require(ERC20Wrapper(wrapper).depositFor(initiator(), amount), ErrorsLib.UNSECCESSFULL_DEPOSIT);
```

File: src/ERC20WrapperBundler.sol [hunk 2]
```diff
@@ -50,6 +51,7 @@ abstract contract ERC20WrapperBundler is BaseBundler {
 
         require(amount != 0, ErrorsLib.ZERO_AMOUNT);
 
-        ERC20Wrapper(wrapper).withdrawTo(account, amount);
+        bool success = ERC20Wrapper(wrapper).withdrawTo(account, amount);
+        require(success, ErrorsLib.UNSECCESSFULL_WITHDRAW);
```
Comment (MathisGD) [inline src/ERC20WrapperBundler.sol:55]:
```suggestion
        require(ERC20Wrapper(wrapper).withdrawTo(account, amount), ErrorsLib.UNSECCESSFULL_WITHDRAW);
```

File: src/libraries/ErrorsLib.sol [hunk 3]
```diff
@@ -38,6 +38,12 @@ library ErrorsLib {
     /// @dev Thrown when an action ends up minting/burning more shares than a given slippage.
     string internal constant SLIPPAGE_EXCEEDED = "slippage exceeded";
 
+    /// @dev Thrown when a call to depositFor fails.
+    string internal constant DEPOSIT_FAILED = "Deposit was unsuccessfull";
```
Comment (Rubilmax) [inline src/libraries/ErrorsLib.sol:42]:
```suggestion
    string internal constant DEPOSIT_FAILED = "deposit failed";
```

**Review by adhusson — APPROVED** [review]

File: src/ERC20WrapperBundler.sol [hunk 4]
```diff
@@ -40,7 +41,8 @@ abstract contract ERC20WrapperBundler is BaseBundler {
 
     /// @notice Burns a number of wrapped tokens and withdraws the corresponding number of underlying tokens.
     /// @dev Initiator must have previously transferred their wrapped tokens to the bundler.
-    /// @dev Assumes that `wrapper` implements the `ERC20Wrapper` interface.
+    /// @dev Assumes that `wrapper` implements the `ERC20Wrapper` interface and that the `withdrawTo`function returns a
```
Comment (MerlinEgalite) [inline src/ERC20WrapperBundler.sol:44]:
```suggestion
    /// @dev Assumes that `wrapper` implements the `ERC20Wrapper` interface and that the `withdrawTo` function returns a
```

File: src/ERC20WrapperBundler.sol [hunk 5]
```diff
@@ -23,7 +23,8 @@ abstract contract ERC20WrapperBundler is BaseBundler {
     /// @dev Wraps tokens on behalf of the initiator to make sure they are able to receive and transfer wrapped tokens.
     /// @dev Wrapped tokens must be transferred to the bundler afterwards to perform additional actions.
     /// @dev Initiator must have previously transferred their tokens to the bundler.
-    /// @dev Assumes that `wrapper` implements the `ERC20Wrapper` interface.
+    /// @dev Assumes that `wrapper` implements the `ERC20Wrapper` interface and that the `depositFor`function returns a
```
Comment (MerlinEgalite) [inline src/ERC20WrapperBundler.sol:26]:
```suggestion
    /// @dev Assumes that `wrapper` implements the `ERC20Wrapper` interface and that the `depositFor` function returns a
```

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

**Review by MerlinEgalite — APPROVED** [review]

File: src/libraries/ErrorsLib.sol [hunk 6]
```diff
@@ -38,6 +38,12 @@ library ErrorsLib {
     /// @dev Thrown when an action ends up minting/burning more shares than a given slippage.
     string internal constant SLIPPAGE_EXCEEDED = "slippage exceeded";
 
+    /// @dev Thrown when a call to depositFor fails.
+    string internal constant DEPOSIT_FAILED = "deposit failed";
+
+    /// @dev Thrown when a call to withdrawTo fails.
+    string internal constant WITHDRAW_FAILED = "withdraw failed";
```
Comment (MathisGD) [inline src/libraries/ErrorsLib.sol:45]: optional 
```suggestion
    /// @dev Thrown when a call to depositFor fails.
    string internal constant DEPOSIT_FOR_FAILED = "deposit for failed";

    /// @dev Thrown when a call to withdrawTo fails.
    string internal constant WITHDRAW_TO_FAILED = "withdraw to failed";
```

**Review by MathisGD — APPROVED** [review]

File: src/libraries/ErrorsLib.sol (hunk 6 above)
Comment (Rubilmax) [reply src/libraries/ErrorsLib.sol:45]: `depositFor` then? But I think "deposit" alone is fine, shorter and cheaper

**Pull request merged**